src/generic/UnivPoly.jl

```diff
@@ -876,6 +876,17 @@ function evaluate(a::S, vars::Vector{S}, vals::Vector{V}) where {S <: UnivPoly{T
    return evaluate(a, varidx, vals)
 end
 
+function (a::Union{MPolyRingElem, UniversalPolyRingElem})(;kwargs...)
+   S = parent(a)
+   vars = Array{Int64}(undef, length(kwargs))
+   vals = Array{RingElement}(undef, length(kwargs))
```

Comment:
```suggestion
   vals = Array{RingElement}(undef, length(kwargs))
```
this uses a rather pessimistic container type, in particular if all the elements have the same type. Probably `vals = [x.second for x in kwargs]` is to prefer.

Reply:
Hm, I'm not sure about that. With `vals = [x.second for x in kwargs]` the result will very quickly become a `Vector{Any}`.

Reply:
Can you give the example where it produces `Vector{Any}` and not `Vector{RingElement}`?

Reply:
```
julia> R,(x,y)=QQ[:x,:y]
(Multivariate polynomial ring in 2 variables over rationals, AbstractAlgebra.Generic.MPoly{Rational{BigInt}}[x, y])

julia> kwargs = (:a => ZZ(1), :b => x)
(:a => 1, :b => x)

julia> vals = [x.second for x in kwargs]
2-element Vector{Any}:
 1
  x

```

Reply:
I see. This is due to `RingElement` being a union. But as this is a union with too many elements, `Vector{RingElement}` will be as slow as `Vector{Any}` anyway.

But we can merge it as is now and add the `vals = [x.second for x in kwargs]` optimization later.